pkg/providers/vault/client.go

```diff
@@ -24,17 +25,39 @@ import (
 	"net/http"
 	"time"
 
+	"github.com/edgexfoundry/go-mod-core-contracts/clients/logger"
+
 	"github.com/edgexfoundry/go-mod-secrets/pkg"
 )
 
+const (
+	AuthTypeHeader = "X-Vault-Token"
+)
+
 // Client defines the behavior for interacting with the Vault REST secret key/value store via HTTP(S).
 type Client struct {
 	HttpConfig SecretConfig
 	HttpCaller Caller
+
+	// internal member variables
+	lc logger.LoggingClient
 }
 
+// package stateful map variable to handle the case of the same caller to have
```

Comment:
What's the specific use case(s) that would have a caller create multiple secret clients (i.e. make multiple calls to `NewSecretClient()`)?

Reply:
I think app service has one use case that a caller to create 2 secret clients, one for database creds and another one for cloud or external creds and potentially they are using different tokens for sure.

Reply:
Is the above a real use case @lenny-intel / @rsdmike?

Reply:
Reached out to Lenny and Mike directly.  Unresolved until I get resolution.

Reply:
> 
> 
> I think app service has one use case that a caller to create 2 secret clients, one for database creds and another one for cloud or external creds and potentially they are using different tokens for sure.

https://github.com/edgexfoundry/go-mod-secrets/pull/45#discussion_r370802012


@lenny-intel  : would you please give some feedback to @michaelestrin 's question above?

Reply:
@jim-wang-intel , done.